### DISCUSSION TITLE
How to install MRiscoC firmware on Ender 3 S1 Pro

### DISCUSSION THREAD

**Discussion by Shany94** [mriscoc/Ender3V2S1#1298] [Show and tell]
### How to install Marlin on Ender 3 S1 Pro:

1. Go to the Creality website and download the firmware
 **Ender-3 S1_Pro_HWv24S1_301_SWV2.0.8.24F4_FDM_LASER dated 2022-09-01.**

Link to Creality firmware download page:
[https://www.crealitycloud.com/software-firmware/firmware/ender-series](https://www.crealitycloud.com/software-firmware/firmware/ender-series)

2. Inside the downloaded Creality firmware folder, you'll find several folders.

You need to update the original display by disconnecting it from its cable, removing the four screws and the back cover, and inserting a micro SD card (formatted in **FAT 32** with **4096 bytes allocation unit**, preferably **4-8 GB**) containing the folders **DWIN_SET** and **private** into the micro SD slot on the display.

3. Reassemble and connect the display to the printer and turn it on; the update should start.

4. Once completed, turn off the printer, remove the micro SD from the display, reassemble it, and connect it to the printer.

5. Now, take the same micro SD with an SD adapter or a different SD card formatted in the same way, cancel all files and instead paste the **STM32F4_UPDATE** folder to update the printer with the original Creality firmware. Insert the SD into the printer, turn it on, and the update should start.

6. Once completed, now that both the screen and printer are updated, replace the original display of the S1 Pro with the one with the knob.

7. To change the screen, you need to open the back cover of the printer to access the motherboard. To do this, turn the printer on its side and unscrew the six small silver screws and the two larger black screws holding the vertical axes. (You only need to unscrew the innermost screw of the left vertical axis and the innermost screw of the right vertical axis.)

8. Disconnect the cable from the original display and connect the rainbow cable of the display with the knob to its slot.

9. Route it through the slot where the original cable was as much as possible; otherwise, it will interfere with the printer's tool drawer.

10. Since the rainbow cable is slightly shorter than the original display cable, it won't reach the original cable exit hole, so you have to route it under the cover, finding the best position to reach the display. It's important not to tighten the screws of the cover excessively in that area to avoid cutting the cable, which is quite resilient fortunately. When you finish the work you can reassemble the back cover.

11. If you haven't already, you need to also disassemble the display with the knob to ensure its type and install the appropriate update for your model. You have all the information you need at this link:

Link to display assets:
[https://github.com/mriscoc/Ender3V2S1/tree/Ender3V2S1-Released/display%20assets](https://github.com/mriscoc/Ender3V2S1/tree/Ender3V2S1-Released/display%20assets)

12. After updating, removing the micro SD from the knob display, and reassembling the display, insert an SD card with MriscoC firmware inside a folder named "**STM32F4_UPDATE**" without changing the file name. Insert your SD card into the printer and turn it on. At this point, the MRiscoC firmware will be installed, and you can enjoy it.

Compatible firmwares are Ender3S1-F4-UBL-20240126.bin and Ender3S1-F4-20240126.bin, downloadable at the bottom of the release page:

Link to firmware release page:
[(https://github.com/mriscoc/Ender3V2S1/releases/tag/20240122](https://github.com/mriscoc/Ender3V2S1/releases/tag/20240122)

**Comment by Despain73** [comment]
Well I am not sure what I did wrong but now I have a blank display it told me my Private was too big needed to be 1mb

Anyone please help me I am looking at a brick currently.  Here are pictures of my boards.  I am desperate Please.
![IMG_6183](https://github.com/user-attachments/assets/fa832625-185a-40a6-8e59-08dd9d92ce49)
![IMG_6188](https://github.com/user-attachments/assets/0f010dcd-a92e-406b-8534-f23987664a0a)
![IMG_6189](https://github.com/user-attachments/assets/f1eb3740-2a92-4986-a3cc-cd8012508257)
